test/parallel_api/iterator/input_data_sweep_usm_allocator.pass.cpp

```diff
@@ -0,0 +1,127 @@
+// -*- C++ -*-
+//===-- input_data_sweep_usm_allocator.pass.cpp ---------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/utils.h"
+#include _PSTL_TEST_HEADER(execution)
+#include _PSTL_TEST_HEADER(algorithm)
+#include _PSTL_TEST_HEADER(iterator)
+
+#include "input_data_sweep.h"
+
+#include "support/utils_invoke.h"
+
+//This test is written without indirection from invoke_on_all_hetero_policies to make clear exactly which types
+// are being tested, and to limit the number of types to be within reason.
+
+#if TEST_DPCPP_BACKEND_PRESENT
+
+template <typename T, int __recurse, typename Policy>
+void
+test(Policy&& policy, T trash, size_t n, const std::string& type_text)
+{
+    if (TestUtils::has_types_support<T>(policy.queue().get_device()))
+    {
+        { //std::vector using usm shared allocator
```

Comment:
I think we need to make sure the kernels for the std::vector with usm shared allocator test end up with a different kernel name than the kernels for usm host allocator test.

I see a duplicate kernel name issue with no unnamed lambda compilation.

Reply:
fixed.  thanks.